### DISCUSSION TITLE
How do I use whispercpp with options from whisper.cpp?

### DISCUSSION THREAD

**Discussion by hb-man** [aarnphm/whispercpp#93] [Q&A] [answered]
Hi, a question about passing options of whisper.cpp [described in its readme](https://github.com/ggerganov/whisper.cpp#quick-start).
Say I want to run it with an option enabling saving transcription to an SRT files   `-osrt`
How do I do that with whispercpp?

**Comment by aarnphm** [comment, marked as answer]
`-osrt` on main.cpp are postprocessing. You can do that in Python separately. 

whisper.cpp provides the binding to the C API, and depends on the programmers to do whatever with it.

**Reply by hb-man** [reply]
Okay, got it.
Than I have a follow up question - how do I produce a transcription with timestamps? From what I tried with `transcribe_from_file()` results in a transcript without timestamps and I can't find options in code to turn timestamps on.

**Reply by aarnphm** [reply]
set `print_realtime` to True should do the job.

**Reply by jake1271** [reply]
A somewhat related question, how to send options like word level timestamps, in whisper.cpp it would be -ml 1
thank you

**Reply by aarnphm** [reply]
Take a look at how it is implemented in the cpp file. All of the options API are 1-to-1 conversion from the C API

**Reply by jake1271** [reply]
Ok, so currently not possible without some coding changes?

**Reply by aarnphm** [reply]
Not sure if I understand your question, but -ml would just be word_length. Again I think you just have to set the option via the whispercpp.api.Params.

**Reply by jake1271** [reply]
Oh, I meant, that it would not be possible to pass that param using the gui app as it is now. 
One would have to make the coding changes to pass the desired params and then recompile the gui app. Unless I'm not understanding how it works.

**Reply by aarnphm** [reply]
What GUI app are you referring to?

**Comment by jake1271** [comment]
Oh, sorry about the confusion! I got the thread mixed up with another project.
But will definitely try this out :)
